File: 10. Subprogram/D. Faktorisasi Prima/d_faktorisasi_prima.java
import java.util.Scanner;

class d_faktorisasi_prima {
    public static boolean isFactor(int N, int x) {
        if (N % x == 0) {
            return true;
        }
        return false;
    }

    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        int N = scan.nextInt();
        scan.close();
        int i, pos, power;
        i = 2;
        pos = 0;
        while (N > 1) {
            power = 0;
            if (isFactor(N, i)) {
                pos++;
                while (N % i == 0) {
                    N /= i;
                    power++;
                }
                if (pos > 1) {
                    System.out.print(" x ");
                }
                System.out.print(i);
                if (power > 1) {
                    System.out.print("^" + power);
                }
            }
            i++;
        }
    }
}

File: 10. Subprogram/D. Faktorisasi Prima/d_faktorisasi_prima.go
package main

import (
  "fmt"
)

func isFactor(N int, x int) bool {
  if N%x == 0 {
    return true
  }
  return false
}

func main() {
  var N int
  fmt.Scan(&N)
  i := 2
  pos := 0
  for N > 1 {
    power := 0
    if isFactor(N, i) {
      pos += 1
      for N%i == 0 {
        N /= i
        power += 1
      }
      if pos > 1 {
        fmt.Print(" x ")
      }
      fmt.Print(i)
      if power > 1 {
        fmt.Printf("^%d", power)
      }
    }
    i += 1
  }
}
